packaging/os/pacman.py

```diff
@@ -130,6 +152,22 @@ def update_package_db(module):
     else:
         module.fail_json(msg="could not update package db")
 
+def upgrade(module):
+    cmdupgrade = "pacman -Suq --noconfirm"
+    cmdneedrefresh = "pacman -Supq"
```

Comment:
This is somewhat a minor implementation point, but using `pacman -Qu` will tell you via the exit code if there are any outdated packages (rather than trying to counting newlines from the parsed output of `pacman -Supq`). It might make the logic a bit cleaner to use that here?